File: hello.rb
#include <iostream>

int main(int argc, char* argv[])
{

name = ARGV.first || "World";
  if (argc < 2)
    {
      std::cerr << "Usage: " << argv[0] << " NAME" << std::endl;
      return 1;
    }
    std::cout << "Hello, " << argv[1] << "!" << std::endl;
    return 0;
}

File: hello.cpp
#include <iostream>

using namespace std;

int main(int argc, char* argv[])
{
  // Default is "World"'
  string  name = "World";
  if (argc == 2)
    cout << "Hello, " << argv[1] << "!" << endl;
  else
    cout << "Hello, " << name << "!" << endl;
  return 0;
}
